api/src/common/exceptions.py

```diff
@@ -44,7 +44,7 @@ def __init__(
         self.extra = extra
         self.severity = severity
 
-    def dict(self) -> dict[str, int | str | dict[str, Any] | None]:
+    def model_dump(self) -> dict[str, int | str | dict[str, Any] | None]:
```

Comment:
The pydantic warning you see when running pytest is not refering to this one.
By adding this to pyproject toml pytest:
```
filterwarnings = [
    "error"
]
```
Warning will raise errors, and you can see where they come from.

So, IMO, no need to rename this function. It's nice to differentiate it from a pydantic model